File: backend/src/038-colorear_stdout/stdout_color.rb
#!/bin/ruby

require "colorize"   # gem install colorize

# Consultar los colores posibles
print String.color_samples

# Consultar los estilos disponibles
print String.modes

puts "Texto verde".colorize(:color => :green)
puts "Texto verde claro".light_green
puts "Texto azul claro".light_blue

puts "Texto en negritas".bold
puts "Texto celeste en cursivas".italic.light_cyan
puts "Texto parpadeante en rojo".blink.red
puts "Texto magenta con fondo amarillo".colorize(:color => :magenta, :background => :light_yellow)

# Fuentes:
# https://github.com/fazibear/colorize

File: backend/src/038-colorear_stdout/stdout_color.js
"use strict"

const colors = require("colors"); // npm install colors

// Para consultar todos los estilos y colores:
console.log(colors.styles);

// Aplicar color
console.log("Texto verde".green)
console.log("Texto rojo".red)

// Aplicar color de fondo
console.log("Texto azul con el fondo blanco".blue.whiteBG)

// Aplicar estilos:
console.log("Texto subrayado".underline)
console.log("Texto amarillo en negrita".yellow.bold)
console.log("Texto cursivo en magenta".magenta.italic)

// Para desabilitar los colores en el script 
// podemos ejecutarlo con la opción --no-color

// Para más opciones como temificar consultar la documentación
// https://github.com/marak/colors.js/